mm2src/coins/utxo/utxo_balance_events.rs

```diff
@@ -172,3 +156,32 @@ impl EventStreamer for UtxoBalanceEventStreamer {
         }
     }
 }
+
+async fn subscribe_to_addresses(
+    utxo: &UtxoCoinFields,
+    addresses: HashSet<Address>,
+) -> Result<BTreeMap<String, Address>, String> {
+    match utxo.rpc_client.clone() {
+        UtxoRpcClientEnum::Electrum(client) => {
+            // Collect the scrpithash for every address into a map.
+            let scripthash_to_address_map = addresses
+                .into_iter()
+                .map(|address| {
+                    let scripthash = address_to_scripthash(&address).map_err(|e| e.to_string())?;
+                    Ok((scripthash, address))
+                })
+                .collect::<Result<HashMap<String, Address>, String>>()?;
+            // Add these subscriptions to the connection manager. It will choose whatever connections
+            // it sees fit to subscribe each of these addresses to.
+            client
+                .connection_manager
+                .add_subscriptions(&scripthash_to_address_map)
+                .await;
```

Comment:
I'm not sure why `add_subscriptions` won't take `BtreeMap` instead of the back and forth conversions.... This might need to be reworked (allow `add_subscriptions` take  a `BtreeMap` instead of HashMap if there's not a good reason not to) if `subscribe_to_addresses` will be invoked many times.

Reply:
thanks, done here [c3b2047](https://github.com/KomodoPlatform/komodo-defi-framework/pull/2431/commits/c3b20474dddb79a5a85f276511ffe40f2b9991c7)